R/plumber-step.R

```diff
@@ -307,6 +315,68 @@ PlumberEndpoint <- R6Class(
       self$path <- path
       private$regex <- createPathRegex(path, self$getFuncParams())
       path
+    },
+    #' @description convert args to specific types
+    #' @param args List of args to convert
+    convertArgs = function(args) {
+      if (!isTRUE(getOption("plumber.typedParameters", TRUE))) {
+        return(args)
+      }
+
+      if (is.null(args)) {
+        # no args to convert
+        return(args)
+      }
+
+      if (length(args) == 0) {
+        # no args to convert
+        return(args)
+      }
+
+      args_names <- names(args)
+      if (is.null(args_names)) {
+        # no names - won't be able to dp any conversions
+        return(args)
+      }
+
+      params <- self$getEndpointParams()
+
+      if (length(params) == 0) {
+        # no information to convert by
+        return(args)
+      }
+
+      # TODO - I suspect someone with more base-R could turn this
+      # for into some wonderful s/lapply thing...
+      converted_args <- list()
+
+      for (i in 1:length(args)) {
+        key <- args_names[[i]]
+        arg <- args[[i]]
+        if (key == "") {
+          # not sure why... but we allow unnamed arguments through
+          converted_args <- c(converted_args, arg)
```

Comment:
> `            # not sure why... but we allow unnamed arguments through`

Do you have an example? Or where did you see this in other code?

I'd understand empty values, but not empty keys.  `?=1` .. doesn't immediately make sense to me as it being valid.